src/Kestrel.Core/Internal/Http/Frame.cs

```diff
@@ -345,11 +345,11 @@ public void Reset()
             StatusCode = StatusCodes.Status200OK;
             ReasonPhrase = null;
 
-            RemoteIpAddress = RemoteEndPoint?.Address;
-            RemotePort = RemoteEndPoint?.Port ?? 0;
+            RemoteIpAddress = HttpConnectionFeature?.RemoteIpAddress;
+            RemotePort = HttpConnectionFeature?.RemotePort ?? 0;
```

Comment:
Also will the `IHttpConnectionFeature` Ip:Ports change from the initial value across resets? Cache locally then set to local values on Reset?